NEWS: Avoid explicitly mentioning freenode in the pinning examples

dequis opened the pull request:
They have proper certs, so using them as an example is wrong.
Particularly worse since they started using letsencrypt recently so
every server has a different cert and pubkey.

We'll figure out how to link this from the release notes later.

ailin-nemui commented:
you can edit the release notes on github releases post factum and you can edit the link on the website. please commit this to a maint/1.0.0 branch

dequis commented:
Done: pushed maint/1.0.0 which currently is the same as 1.0.0, rebased this PR to that, changed the base in github to that, updated github release notes with the same thing as this.

Next: Merging this commit to that branch, and the release notes can be changed to point to NEWS in maint/1.0.0 instead of the 1.0.0 tag. Ack?

ailin-nemui commented:
Yes you can link to https://github.com/irssi/irssi/blob/maint/1.0.0/NEWS on the website.

dequis commented:
Done https://github.com/irssi/irssi.github.io/commit/3bfa09e5fe53e0521ca4c775b62aea4de234960d

https://irssi.org/2017/01/05/irssi-1.0.0-released/
